### ISSUE TITLE
Invalid Scopes: read_stream, user_groups. 

### ISSUE THREAD

**Issue by pcmanpa** [jeckman/wpbook-lite#1]
I get the following error when I click the Generate Access Token link on the settings page. I am running WordPress 4.2.4 and WPBook Lite version 1.6.5. Thank you

Invalid Scopes: read_stream, user_groups. This message is only shown to developers. Users of your app will ignore these permissions if present. Please read the documentation for valid permissions at: https://developers.facebook.com/docs/facebook-login/permissions

I tried starting over from scratch and still get the same error.

https://wordpress.org/plugins/wpbook-lite/

**Comment by jeckman** [comment]
Looks like Facebook API has deprecated read_stream and user_groups permissions. 

See https://www.webniraj.com/2015/07/14/facebook-api-graph-api-v2-4-released-removes-groups-notifications-and-stream-permissions/  and https://developers.facebook.com/docs/apps/changelog#v2_4

Removing the ask for those permissions is a pretty simple task, but it isn't clear to me (it's been a while since I've been in this code) what impact removing them will have. 

In other words, if we generate an access token that no longer includes read_stream, will we still be able to read the stream? 

Will require some testing before release. 

@pcmanpa  are you able to install a beta version and test before I release to the WordPress.org repo?

**Comment by jeckman** [comment]
Just updated

**Comment by pcmanpa** [comment]
Yes I can test a beta for sure. I'm eager to use this plugin. Just let me know when and what you'd like me to do. Thank you

**Comment by jeckman** [comment]
@pcmanpa  if you can download the current zip from here on github and install it and test it that would be helpful. 

I just tested on one of my sites and it worked fine for generating an access token, but I haven't tested cross posting or importing comments yet.

**Comment by pcmanpa** [comment]
OK it did indeed save the access token. I will try and test a post as well. Thank you

**Comment by pcmanpa** [comment]
When I posted an item i received the following

No post id returned from Facebook, $fb_response was /n and $fb_page_type was and $wpbook_description

**Comment by pcmanpa** [comment]
On a side note it did post to FB but it doesn't seem comments are working. Thank you

**Comment by jeckman** [comment]
Hmm. I'm afraid that may mean that if there is no longer a read_stream permission wpbook-lite won't be able to read the user's stream, which is what is necessary to be able to get comments from the items cross-posted.  

I will continue to look into it as time allows

**Comment by pcmanpa** [comment]
Ok thank you. Much appreciated.

**Comment by jeckman** [comment]
Just added a user_posts permission, which may fix this. You'd need to download a new version from github, install on your blog, and re-do the "grant permissions" flow - it will now ask for user_posts which should enable it to still read your stream. 

(Note you'd have to make a new post, cross-posted to FB - it won't automagically be able to import comments on a post already posted - the one where you got no post_id back from FB)

**Comment by pcmanpa** [comment]
I now get the following error and it doesn't post.

No post id returned from Facebook, $fb_response was /n and $fb_page_type was and $wpbook_description was and $my_title was wpbook test 2

**Comment by holls91** [comment]
I'm trying to setup WPBook-Lite and getting the same error as the first post (Invalid Scopes: read_stream, user_groups).
Will it be any fix?
Thanks,
Simona
